test/jdk/java/awt/MenuItem/LotsOfMenuItemsTest.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+
+import java.util.List;
+
+/*
+ * @test
+ * @bug 4175790
+ * @requires os.family == "windows"
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Win32: Running out of command ids for menu items
+ * @run main/manual LotsOfMenuItemsTest
+ */
+
+public class LotsOfMenuItemsTest extends ComponentAdapter {
+    private static final int NUM_WINDOWS = 400;
+    private static TestFrame firstFrame;
+    private static TestFrame testFrame;
+
+    public static void main(String[] args) throws Exception {
+        LotsOfMenuItemsTest obj = new LotsOfMenuItemsTest();
+        String INSTRUCTIONS = """
+                This test creates lots of frames with menu bars.
+                When it's done you will see two frames.
+                Try to select menu items from each of them.
+
+                If everything seems to work - test passed.
+                Click "Pass" button in the test harness window.
+
+                If test crashes on you - test failed.""";
+
+        PassFailJFrame.builder()
+                .title("LotsOfMenuItemsTest")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(obj.createAndShowUI())
+                .build()
+                .awaitAndCheck();
+    }
+
+    private List<Frame> createAndShowUI() {
+        firstFrame = new TestFrame("First frame");
+        firstFrame.addComponentListener(this);
+
+        for (int i = 1; i < NUM_WINDOWS - 1; ++i) {
+            testFrame = new TestFrame("Running(" + i + ")...");
+            testFrame.setVisible(false);
+            testFrame.dispose();
+        }
+        testFrame = new TestFrame("Last Frame");
+        return List.of(firstFrame, testFrame);
+    }
+
+    public void componentShown(ComponentEvent e) {
```

Comment:
```suggestion
    @Override
    public void componentShown(ComponentEvent e) {
```
Please add `@Override` annotation where applicable. It makes it explicit that a method implements an interface or overrides a method of a superclass.

Reply:
Updated.